### ISSUE TITLE
Propagation of certificates after acme-* renewal?

### ISSUE THREAD

**Issue by calbrecht** [nix-community/authentik-nix#12]
With setting `nginx.enableAcme = true;` on the day when initially discovered certificates expire, a configured ldap provider cannot be queried any longer because of outdated certificates.

In the admin GUI, we had to manually:
* delete the discovered certificate (/admin/#/crypto/certificates)
* run the certificate_discovery task (/admin/#/administration/system-tasks)
* reassign the certificate to the tenant (/admin/#/core/tenants -> edit -> Other global settings)
* reassign the certificate to the provider (/admin/#/core/providers -> edit -> Protocol settings)

This hit us by surprise, by any chance, do you see a way to automate this process? 
At least i guess it would be worth of a warning in the README, right?

**Comment by WilliButz** [comment]
Was the expired certificate managed by authentik or imported manually at some point? For the latter, I'd expect this behavior.
And for the former, I cannot see the same behavior for a Let's Encrypt certificate which was recently renewed and automatically discovered by authentik:
![discovered_cert](https://github.com/nix-community/authentik-nix/assets/20464732/fb8c0e11-643e-4d3d-aa2b-16a40cf0b5cb)

(note "Managed by authentik (Discovered)")

See also https://goauthentik.io/docs/core/certificates#external-certificates
> Files are checked every 5 minutes, and will trigger an Outpost refresh if the files differ.

**Comment by calbrecht** [comment]
Mysterious, the one that wasn't changed is a discovered one as well. 

There is still a longer living self-signed certificate beneath, and it might be that authentik wasn't restarted after the certificate was refreshed, beside after detecting the outdate causing ldap service disruption. Even the restart wasn't effective although we didn't wait for 5 minutes.

I will keep an eye on the next change cycle to see if it will happen again.

**Comment by calbrecht** [comment]
In the current cycle authentik correctly noticed the replacement of the certificate. The reason why it didn't remains unresolved.

**Comment by Ma27** [comment]
Agreed. It's a bummer that systemd doesn't refresh the credential dir in at least some cases, but this is a reasonable workaround.
